ultravox/data/datasets.py

```diff
@@ -1153,17 +1207,34 @@ def __iter__(self):
         return (self._process(sample) for sample in self._dataset)
 
 
-class Range(data.IterableDataset):
+class Range(SizedIterableDataset):
     """Limits the number of samples from another dataset."""
 
     def __init__(
-        self, dataset: data.IterableDataset, num_samples: Optional[int] = None
+        self, dataset: SizedIterableDataset, num_samples: Optional[int] = None
```

Comment:
What's the reason to allow `num_samples = None`? Conceptually, I thought `num_samples` should always be nonnegative and no greater than `len(self._dataset)`

Reply:
This allows using `Range`, not just when `num_samples` is provided, but always. So the options are: the full dataset or an artificially restricted dataset. So `Range(ds, num_samples=None)` is the same as `ds`.

Reply:
I see there are places we don't check `num_samples` before using `Range`.